src/lib/operators/aggregate_hash.cpp

```diff
@@ -316,15 +348,74 @@ void AggregateHash::_aggregate() {
                     keys_per_chunk[chunk_id][chunk_offset][group_column_index] = 0u;
                   }
                 } else {
-                  auto inserted = id_map.try_emplace(position.value(), id_counter);
+                  auto id = AggregateKeyEntry{0} - 1;  // using max value for "unset"
```

Comment:
We need to generate an ID that is unique for the value. In some cases, we can use an optimization,
in others, we can't. We need to somehow track whether we have found an ID or not. For this, we
first set `id` to its maximum value. If after all branches it is still that max value, no
optimized ID generation was applied and we need to generate the ID using the value->ID map.

Does this help?

Reply:
Did not realize that comment was still lurking around (started the review a while ago). I think a comment would definitely help here, but it can be short (the `id` becomes pretty clear when reading further, sorry for the confusing question).

What the next question was about is the overflow. I'd rather use `numeric_limits<uint>(max)` if possible.

Reply:
done